axon/act_prjn.go

```diff
@@ -126,8 +127,42 @@ func (sc *SynComParams) ReadOff(cycTot int32) uint32 {
 // ReadIdx returns index for reading existing spikes from the GBuf buffer,
 // based on the layer-based recv neuron index and the
 // ReadOff offset from the CycleTot.
-func (sc *SynComParams) ReadIdx(rnIdx uint32, cycTot int32) uint32 {
-	return rnIdx*sc.DelLen + sc.ReadOff(cycTot)
+func (sc *SynComParams) ReadIdx(rnIdx uint32, cycTot int32, nRecvNeurs uint32) uint32 {
+	// return rnIdx*sc.DelLen + sc.ReadOff(cycTot)
+	return sc.ReadOff(cycTot)*nRecvNeurs + rnIdx // delay is outer, neurs are inner -- should be faster?
+}
+
+// FloatToIntFactor returns the factor used for converting float32
+// to int32 in GBuf encoding.  Because total G is constrained via
+// scaling factors to be around ~1, it is safe to use a factor that
+// uses most of the available bits, leaving enough room to prevent
+// overflow when adding together the different vals.
+// For encoding, bake this into scale factor in SendSpike, and
+// cast the result to int32.
+func (sc *SynComParams) FloatToIntFactor() float32 {
+	return float32(1 << 24) // leaves 7 bits = 128 to cover any extreme cases
```

Comment:
How did you pick the scale factor to be 24? If the value is always between 0 and 1, why not just multiply the float by 2^32-1 (aka MAX_INT)?

Reply:
If the 24 is indeed sort-of arbitrary, and we should clarify that in the comment. If we don't need the precision, we should think about going down to smaller dtypes.

Reply:
this gives 7 bits = 127 x overhead in case values are unexpectedly large, and 24 bits gives float-level precision in tests.

Reply:
this already has all the relevant comments in place.  you can feel free to add more but it conveys everything I would want to say already.